diff --git a/core/cva5_config.sv b/core/cva5_config.sv
index c7d5550..19d71d5 100755
--- a/core/cva5_config.sv
+++ b/core/cva5_config.sv
@@ -149,19 +149,19 @@ package cva5_config;
     localparam cpu_config_t EXAMPLE_CONFIG = '{
         //ISA options
         INCLUDE_M_MODE : 1,
-        INCLUDE_S_MODE : 1,
-        INCLUDE_U_MODE : 1,
+        INCLUDE_S_MODE : 0,
+        INCLUDE_U_MODE : 0,
         INCLUDE_MUL : 1,
-        INCLUDE_DIV : 1,
-        INCLUDE_IFENCE : 1,
-        INCLUDE_CSRS : 1,
+        INCLUDE_DIV : 0,
+        INCLUDE_IFENCE : 0,
+        INCLUDE_CSRS : 0,
         INCLUDE_AMO : 0,
         //CSR constants
         CSRS : '{
             MACHINE_IMPLEMENTATION_ID : 0,
             CPU_ID : 0,
-            RESET_VEC : 32'h80000000,
-            RESET_MTVEC : 32'h80000100,
+            RESET_VEC : 32'h00000000,
+            RESET_MTVEC : 32'h00000100,
             NON_STANDARD_OPTIONS : '{
                 COUNTER_W : 33,
                 MCYCLE_WRITEABLE : 1,
@@ -205,22 +205,22 @@ package cva5_config;
         },
         INCLUDE_ILOCAL_MEM : 1,
         ILOCAL_MEM_ADDR : '{
-            L : 32'h80000000, 
-            H : 32'h8FFFFFFF
+            L : 32'h00000000, 
+            H : 32'h0FFFFFFF
         },
         INCLUDE_DLOCAL_MEM : 1,
         DLOCAL_MEM_ADDR : '{
-            L : 32'h80000000,
-            H : 32'h8FFFFFFF
+            L : 32'h00000000,
+            H : 32'h0FFFFFFF
         },
         INCLUDE_PERIPHERAL_BUS : 1,
         PERIPHERAL_BUS_ADDR : '{
-            L : 32'h60000000,
-            H : 32'h6FFFFFFF
+            L : 32'h11000000,
+            H : 32'h11FFFFFF
         },
         PERIPHERAL_BUS_TYPE : AXI_BUS,
         //Branch Predictor Options
-        INCLUDE_BRANCH_PREDICTOR : 1,
+        INCLUDE_BRANCH_PREDICTOR : 0,
         BP : '{
             WAYS : 2,
             ENTRIES : 512,
@@ -285,7 +285,7 @@ package cva5_config;
     ////////////////////////////////////////////////////
     //Trace Options
     //Trace interface is necessary for verilator simulation
-    localparam ENABLE_TRACE_INTERFACE = 1;
+    localparam ENABLE_TRACE_INTERFACE = 0;
 
 
     ////////////////////////////////////////////////////
diff --git a/core/cva5_wrapper.sv b/core/cva5_wrapper.sv
new file mode 100644
index 0000000..0cb6a56
--- /dev/null
+++ b/core/cva5_wrapper.sv
@@ -0,0 +1,228 @@
+/*
+ * Copyright © 2022 Carsten Heinz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module cva5_wrapper
+    import cva5_config::*;
+    import l2_config_and_types::*;
+    import riscv_types::*;
+    import cva5_types::*;
+ (
+	input logic clk,
+	input logic rst,
+
+	output wire [29:0] instruction_bram_addr,
+	output wire        instruction_bram_en,
+	output wire [3:0]  instruction_bram_we,
+	output wire [31:0] instruction_bram_din,
+	input  wire [31:0] instruction_bram_dout,
+
+	output wire [29:0] data_bram_addr,
+	output wire        data_bram_en,
+	output wire [3:0]  data_bram_we,
+	output wire [31:0] data_bram_din,
+	input  wire [31:0] data_bram_dout,
+
+	// AXI Bus
+	// AXI Write Channels
+	output wire                            m_axi_awvalid,
+	input  wire                            m_axi_awready,
+	output wire [5:0]                      m_axi_awid,
+	output wire [31:0]                     m_axi_awaddr,
+	//~ output wire [3:0]                      m_axi_awregion,
+	output wire [7:0]                      m_axi_awlen,
+	output wire [2:0]                      m_axi_awsize,
+	output wire [1:0]                      m_axi_awburst,
+	//~ output wire                            m_axi_awlock,
+	output wire [3:0]                      m_axi_awcache,
+	//~ output wire [2:0]                      m_axi_awprot,
+	//~ output wire [3:0]                      m_axi_awqos,
+
+	output wire                            m_axi_wvalid,
+	input  wire                            m_axi_wready,
+	output wire [31:0]                     m_axi_wdata,
+	output wire [3:0]                      m_axi_wstrb,
+	output wire                            m_axi_wlast,
+
+	input  wire                            m_axi_bvalid,
+	output wire                            m_axi_bready,
+	input  wire [1:0]                      m_axi_bresp,
+	input  wire [5:0]                      m_axi_bid,
+
+	// AXI Read Channels
+	output wire                            m_axi_arvalid,
+	input  wire                            m_axi_arready,
+	output wire [5:0]                      m_axi_arid,
+	output wire [31:0]                     m_axi_araddr,
+	//~ output wire [3:0]                      m_axi_arregion,
+	output wire [7:0]                      m_axi_arlen,
+	output wire [2:0]                      m_axi_arsize,
+	output wire [1:0]                      m_axi_arburst,
+	//~ output wire                            m_axi_arlock,
+	output wire [3:0]                      m_axi_arcache,
+	//~ output wire [2:0]                      m_axi_arprot,
+	//~ output wire [3:0]                      m_axi_arqos,
+
+	input  wire                            m_axi_rvalid,
+	output wire                            m_axi_rready,
+	input  wire [5:0]                      m_axi_rid,
+	input  wire [31:0]                     m_axi_rdata,
+	input  wire [1:0]                      m_axi_rresp,
+	input  wire                            m_axi_rlast,
+
+	// AXI Cache
+	// AXI Write Channels
+	output wire                            axi_awvalid,
+	input  wire                            axi_awready,
+	output wire [5:0]                      axi_awid,
+	output wire [31:0]                     axi_awaddr,
+	//~ output wire [3:0]                      axi_awregion,
+	output wire [7:0]                      axi_awlen,
+	output wire [2:0]                      axi_awsize,
+	output wire [1:0]                      axi_awburst,
+	//~ output wire                            axi_awlock,
+	output wire [3:0]                      axi_awcache,
+	output wire [2:0]                      axi_awprot,
+	//~ output wire [3:0]                      axi_awqos,
+
+	output wire                            axi_wvalid,
+	input  wire                            axi_wready,
+	output wire [31:0]                     axi_wdata,
+	output wire [3:0]                      axi_wstrb,
+	output wire                            axi_wlast,
+
+	input  wire                            axi_bvalid,
+	output wire                            axi_bready,
+	input  wire [1:0]                      axi_bresp,
+	input  wire [5:0]                      axi_bid,
+
+	// AXI Read Channels
+	output wire                            axi_arvalid,
+	input  wire                            axi_arready,
+	output wire [5:0]                      axi_arid,
+	output wire [31:0]                     axi_araddr,
+	//~ output wire [3:0]                      axi_arregion,
+	output wire [7:0]                      axi_arlen,
+	output wire [2:0]                      axi_arsize,
+	output wire [1:0]                      axi_arburst,
+	//~ output wire                            axi_arlock,
+	output wire [3:0]                      axi_arcache,
+	output wire [2:0]                      axi_arprot,
+	//~ output wire [3:0]                      axi_arqos,
+
+	input  wire                            axi_rvalid,
+	output wire                            axi_rready,
+	input  wire [5:0]                      axi_rid,
+	input  wire [31:0]                     axi_rdata,
+	input  wire [1:0]                      axi_rresp,
+	input  wire                            axi_rlast,
+
+	input logic timer_interrupt,
+	input logic interrupt
+);
+
+local_memory_interface instruction_bram();
+assign instruction_bram_addr = {instruction_bram.addr[27:0],2'b0};
+assign instruction_bram_en = instruction_bram.en;
+assign instruction_bram_we = instruction_bram.be;
+assign instruction_bram_din = instruction_bram.data_in;
+assign instruction_bram.data_out = instruction_bram_dout;
+
+local_memory_interface data_bram();
+assign data_bram_addr = {data_bram.addr[27:0],2'b0};
+assign data_bram_en = data_bram.en;
+assign data_bram_we = data_bram.be;
+assign data_bram_din = data_bram.data_in;
+assign data_bram.data_out = data_bram_dout;
+
+axi_interface m_axi();
+assign m_axi_awvalid = m_axi.awvalid;
+assign m_axi.awready = m_axi_awready;
+assign m_axi_awid = m_axi.awid;
+assign m_axi_awaddr = m_axi.awaddr;
+assign m_axi_awlen = m_axi.awlen;
+assign m_axi_awsize = m_axi.awsize;
+assign m_axi_awburst = m_axi.awburst;
+assign m_axi_awcache = m_axi.awcache;
+
+assign m_axi_wvalid = m_axi.wvalid;
+assign m_axi.wready = m_axi_wready;
+assign m_axi_wdata = m_axi.wdata;
+assign m_axi_wstrb = m_axi.wstrb;
+assign m_axi_wlast = m_axi.wlast;
+
+assign m_axi.bvalid = m_axi_bvalid;
+assign m_axi_bready = m_axi.bready;
+assign m_axi.bresp = m_axi_bresp;
+assign m_axi.bid = m_axi_bid;
+
+assign m_axi_arvalid = m_axi.arvalid;
+assign m_axi.arready = m_axi_arready;
+assign m_axi_arid = m_axi.arid;
+assign m_axi_araddr = m_axi.araddr;
+assign m_axi_arlen = m_axi.arlen;
+assign m_axi_arsize = m_axi.arsize;
+assign m_axi_arburst = m_axi.arburst;
+assign m_axi_arcache = m_axi.arcache;
+
+assign m_axi.rvalid = m_axi_rvalid;
+assign m_axi_rready = m_axi.rready;
+assign m_axi.rid = m_axi_rid;
+assign m_axi.rdata = m_axi_rdata;
+assign m_axi.rresp = m_axi_rresp;
+assign m_axi.rlast = m_axi_rlast;
+
+avalon_interface m_avalon();
+wishbone_interface m_wishbone();
+
+l2_requester_interface l2[L2_NUM_PORTS-1:0]();
+assign l2[1].request_push = 0;
+assign l2[1].wr_data_push = 0;
+assign l2[1].inv_ack = l2[1].inv_valid;
+assign l2[1].rd_data_ack = l2[1].rd_data_valid;
+
+l2_memory_interface arb_mem();
+
+cva5 cva5 (
+	.clk(clk),
+	.rst(rst),
+
+	.instruction_bram(instruction_bram),
+	.data_bram(data_bram),
+
+	.m_axi(m_axi),
+	.m_avalon(m_avalon),
+	.m_wishbone(m_wishbone),
+	.l2(l2[0]),
+
+	.s_interrupt(0),
+	.m_interrupt(0)
+);
+
+l2_arbiter l2_arb (
+	.clk(clk),
+	.rst(rst),
+	.request(l2),
+	.mem(arb_mem)
+);
+
+axi_to_arb l2_to_mem (
+	.clk(clk),
+	.rst(rst),
+	.l2(arb_mem),
+	.* // e.g. axi_*
+);
+
+endmodule
diff --git a/core/cva5_wrapper_verilog.v b/core/cva5_wrapper_verilog.v
new file mode 100644
index 0000000..944ebef
--- /dev/null
+++ b/core/cva5_wrapper_verilog.v
@@ -0,0 +1,267 @@
+/*
+ * Copyright © 2022 Carsten Heinz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module cva5_wrapper_verilog (
+	input wire clk,
+	input wire rst,
+
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 instruction_bram CLK" *)
+	output wire        instruction_bram_clk,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 instruction_bram RST" *)
+	output wire        instruction_bram_rst,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 instruction_bram ADDR" *)
+	output wire [29:0] instruction_bram_addr,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 instruction_bram EN" *)
+	output wire        instruction_bram_en,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 instruction_bram WE" *)
+	output wire [3:0]  instruction_bram_we,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 instruction_bram DIN" *)
+	output wire [31:0] instruction_bram_din,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 instruction_bram DOUT" *)
+	input  wire [31:0] instruction_bram_dout,
+
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 data_bram CLK" *)
+	output wire        data_bram_clk,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 data_bram RST" *)
+	output wire        data_bram_rst,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 data_bram ADDR" *)
+	output wire [29:0] data_bram_addr,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 data_bram EN" *)
+	output wire        data_bram_en,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 data_bram WE" *)
+	output wire [3:0]  data_bram_we,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 data_bram DIN" *)
+	output wire [31:0] data_bram_din,
+	(* X_INTERFACE_INFO = "xilinx.com:interface:bram:1.0 data_bram DOUT" *)
+	input  wire [31:0] data_bram_dout,
+
+	// AXI Bus
+	// AXI Write Channels
+	output wire                            m_axi_awvalid,
+	input  wire                            m_axi_awready,
+	//~ output wire [5:0]                      m_axi_awid,
+	output wire [31:0]                     m_axi_awaddr,
+	//~ output wire [3:0]                      m_axi_awregion,
+	output wire [7:0]                      m_axi_awlen,
+	output wire [2:0]                      m_axi_awsize,
+	output wire [1:0]                      m_axi_awburst,
+	//~ output wire                            m_axi_awlock,
+	output wire [3:0]                      m_axi_awcache,
+	//~ output wire [2:0]                      m_axi_awprot,
+	//~ output wire [3:0]                      m_axi_awqos,
+
+	output wire                            m_axi_wvalid,
+	input  wire                            m_axi_wready,
+	output wire [31:0]                     m_axi_wdata,
+	output wire [3:0]                      m_axi_wstrb,
+	output wire                            m_axi_wlast,
+
+	input  wire                            m_axi_bvalid,
+	output wire                            m_axi_bready,
+	input  wire [1:0]                      m_axi_bresp,
+	//~ input  wire [5:0]                      m_axi_bid,
+
+	// AXI Read Channels
+	output wire                            m_axi_arvalid,
+	input  wire                            m_axi_arready,
+	//~ output wire [5:0]                      m_axi_arid,
+	output wire [31:0]                     m_axi_araddr,
+	//~ output wire [3:0]                      m_axi_arregion,
+	output wire [7:0]                      m_axi_arlen,
+	output wire [2:0]                      m_axi_arsize,
+	output wire [1:0]                      m_axi_arburst,
+	//~ output wire                            m_axi_arlock,
+	output wire [3:0]                      m_axi_arcache,
+	//~ output wire [2:0]                      m_axi_arprot,
+	//~ output wire [3:0]                      m_axi_arqos,
+
+	input  wire                            m_axi_rvalid,
+	output wire                            m_axi_rready,
+	//~ input  wire [5:0]                      m_axi_rid,
+	input  wire [31:0]                     m_axi_rdata,
+	input  wire [1:0]                      m_axi_rresp,
+	input  wire                            m_axi_rlast,
+
+	// AXI Cache
+	// AXI Write Channels
+	output wire                            m_axi_cache_awvalid,
+	input  wire                            m_axi_cache_awready,
+	output wire [5:0]                      m_axi_cache_awid,
+	output wire [31:0]                     m_axi_cache_awaddr,
+	//~ output wire [3:0]                      m_axi_cache_awregion,
+	output wire [7:0]                      m_axi_cache_awlen,
+	output wire [2:0]                      m_axi_cache_awsize,
+	output wire [1:0]                      m_axi_cache_awburst,
+	//~ output wire                            m_axi_cache_awlock,
+	output wire [3:0]                      m_axi_cache_awcache,
+	output wire [2:0]                      m_axi_cache_awprot,
+	//~ output wire [3:0]                      m_axi_cache_awqos,
+
+	output wire                            m_axi_cache_wvalid,
+	input  wire                            m_axi_cache_wready,
+	output wire [31:0]                     m_axi_cache_wdata,
+	output wire [3:0]                      m_axi_cache_wstrb,
+	output wire                            m_axi_cache_wlast,
+
+	input  wire                            m_axi_cache_bvalid,
+	output wire                            m_axi_cache_bready,
+	input  wire [1:0]                      m_axi_cache_bresp,
+	input  wire [5:0]                      m_axi_cache_bid,
+
+	// AXI Read Channels
+	output wire                            m_axi_cache_arvalid,
+	input  wire                            m_axi_cache_arready,
+	output wire [5:0]                      m_axi_cache_arid,
+	output wire [31:0]                     m_axi_cache_araddr,
+	//~ output wire [3:0]                      m_axi_cache_arregion,
+	output wire [7:0]                      m_axi_cache_arlen,
+	output wire [2:0]                      m_axi_cache_arsize,
+	output wire [1:0]                      m_axi_cache_arburst,
+	//~ output wire                            m_axi_cache_arlock,
+	output wire [3:0]                      m_axi_cache_arcache,
+	output wire [2:0]                      m_axi_cache_arprot,
+	//~ output wire [3:0]                      m_axi_cache_arqos,
+
+	input  wire                            m_axi_cache_rvalid,
+	output wire                            m_axi_cache_rready,
+	input  wire [5:0]                      m_axi_cache_rid,
+	input  wire [31:0]                     m_axi_cache_rdata,
+	input  wire [1:0]                      m_axi_cache_rresp,
+	input  wire                            m_axi_cache_rlast
+);
+
+assign instruction_bram_clk = clk;
+assign instruction_bram_rst = rst;
+assign data_bram_clk = clk;
+assign data_bram_rst = rst;
+
+wire [5:0]                      m_axi_awid;
+wire [5:0]                      m_axi_bid;
+wire [5:0]                      m_axi_arid;
+wire [5:0]                      m_axi_rid;
+
+
+cva5_wrapper cva5_wrapper (
+	.clk(clk),
+	.rst(rst),
+
+	.instruction_bram_addr(instruction_bram_addr),
+	.instruction_bram_en(instruction_bram_en),
+	.instruction_bram_we(instruction_bram_we),
+	.instruction_bram_din(instruction_bram_din),
+	.instruction_bram_dout(instruction_bram_dout),
+
+	.data_bram_addr(data_bram_addr),
+	.data_bram_en(data_bram_en),
+	.data_bram_we(data_bram_we),
+	.data_bram_din(data_bram_din),
+	.data_bram_dout(data_bram_dout),
+
+	// AXI Bus
+	.m_axi_awvalid(m_axi_awvalid),
+	.m_axi_awready(m_axi_awready),
+	.m_axi_awid(m_axi_awid),
+	.m_axi_awaddr(m_axi_awaddr),
+	//~ .m_axi_awregion(m_axi_awregion),
+	.m_axi_awlen(m_axi_awlen),
+	.m_axi_awsize(m_axi_awsize),
+	.m_axi_awburst(m_axi_awburst),
+	//~ .m_axi_awlock(m_axi_awlock),
+	.m_axi_awcache(m_axi_awcache),
+	//~ .m_axi_awprot(m_axi_awprot),
+	//~ .m_axi_awqos(m_axi_awqos),
+
+	.m_axi_wvalid(m_axi_wvalid),
+	.m_axi_wready(m_axi_wready),
+	.m_axi_wdata(m_axi_wdata),
+	.m_axi_wstrb(m_axi_wstrb),
+	.m_axi_wlast(m_axi_wlast),
+
+	.m_axi_bvalid(m_axi_bvalid),
+	.m_axi_bready(m_axi_bready),
+	.m_axi_bresp(m_axi_bresp),
+	.m_axi_bid(m_axi_bid),
+
+	.m_axi_arvalid(m_axi_arvalid),
+	.m_axi_arready(m_axi_arready),
+	.m_axi_arid(m_axi_arid),
+	.m_axi_araddr(m_axi_araddr),
+	//~ .m_axi_arregion(m_axi_arregion),
+	.m_axi_arlen(m_axi_arlen),
+	.m_axi_arsize(m_axi_arsize),
+	.m_axi_arburst(m_axi_arburst),
+	//~ .m_axi_arlock(m_axi_arlock),
+	.m_axi_arcache(m_axi_arcache),
+	//~ .m_axi_arprot(m_axi_arprot),
+	//~ .m_axi_arqos(m_axi_arqos),
+
+	.m_axi_rvalid(m_axi_rvalid),
+	.m_axi_rready(m_axi_rready),
+	.m_axi_rid(m_axi_rid),
+	.m_axi_rdata(m_axi_rdata),
+	.m_axi_rresp(m_axi_rresp),
+	.m_axi_rlast(m_axi_rlast),
+
+	// AXI Cache
+	.axi_awvalid(m_axi_cache_awvalid),
+	.axi_awready(m_axi_cache_awready),
+	.axi_awid(m_axi_cache_awid),
+	.axi_awaddr(m_axi_cache_awaddr),
+	//~ .axi_awregion(m_axi_cache_awregion),
+	.axi_awlen(m_axi_cache_awlen),
+	.axi_awsize(m_axi_cache_awsize),
+	.axi_awburst(m_axi_cache_awburst),
+	//~ .axi_awlock(m_axi_cache_awlock),
+	.axi_awcache(m_axi_cache_awcache),
+	.axi_awprot(m_axi_cache_awprot),
+	//~ .axi_awqos(m_axi_cache_awqos),
+
+	.axi_wvalid(m_axi_cache_wvalid),
+	.axi_wready(m_axi_cache_wready),
+	.axi_wdata(m_axi_cache_wdata),
+	.axi_wstrb(m_axi_cache_wstrb),
+	.axi_wlast(m_axi_cache_wlast),
+
+	.axi_bvalid(m_axi_cache_bvalid),
+	.axi_bready(m_axi_cache_bready),
+	.axi_bresp(m_axi_cache_bresp),
+	.axi_bid(m_axi_cache_bid),
+
+	.axi_arvalid(m_axi_cache_arvalid),
+	.axi_arready(m_axi_cache_arready),
+	.axi_arid(m_axi_cache_arid),
+	.axi_araddr(m_axi_cache_araddr),
+	//~ .axi_arregion(m_axi_cache_arregion),
+	.axi_arlen(m_axi_cache_arlen),
+	.axi_arsize(m_axi_cache_arsize),
+	.axi_arburst(m_axi_cache_arburst),
+	//~ .axi_arlock(m_axi_cache_arlock),
+	.axi_arcache(m_axi_cache_arcache),
+	.axi_arprot(m_axi_cache_arprot),
+	//~ .axi_arqos(m_axi_cache_arqos),
+
+	.axi_rvalid(m_axi_cache_rvalid),
+	.axi_rready(m_axi_cache_rready),
+	.axi_rid(m_axi_cache_rid),
+	.axi_rdata(m_axi_cache_rdata),
+	.axi_rresp(m_axi_cache_rresp),
+	.axi_rlast(m_axi_cache_rlast),
+
+	.timer_interrupt(0),
+	.interrupt(0)
+);
+
+endmodule
diff --git a/core/writeback.sv b/core/writeback.sv
index 29d39ca..ea404a2 100644
--- a/core/writeback.sv
+++ b/core/writeback.sv
@@ -58,7 +58,7 @@ module writeback
 
     function unit_count_t get_cumulative_unit_count();
         unit_count_t counts;
-        int unsigned cumulative_count = 0;
+        automatic int unsigned cumulative_count = 0;
         for (int i = 0; i < CONFIG.NUM_WB_GROUPS; i++) begin
             counts[i] = cumulative_count;
             cumulative_count += NUM_UNITS[i];
diff --git a/package.tcl b/package.tcl
new file mode 100644
index 0000000..ad3acd4
--- /dev/null
+++ b/package.tcl
@@ -0,0 +1,38 @@
+puts "package OpenHW Group cva5"
+
+set name cva5
+set version 0.1
+create_project -in_memory
+
+#~ add_files [glob debug_module/*.sv]
+add_files [glob l2_arbiter/*.sv]
+#~ add_files [glob local_memory/*.sv]
+add_files [glob local_memory/local_memory_interface.sv]
+add_files [glob core/lutrams/*.sv]
+add_files [glob core/*{.sv,.v}]
+
+remove_files core/binary_occupancy.sv
+remove_files core/placer_randomizer.sv
+
+update_compile_order -fileset sources_1
+set_property top cva5_wrapper_verilog [current_fileset]
+update_compile_order -fileset sources_1
+
+ipx::package_project -root_dir [pwd]/cva5 -import_files -force -force_update_compile_order
+set core [ipx::current_core]
+set_property vendor openhwgroup $core
+set_property library $name $core
+set_property name $name $core
+set_property display_name $name $core
+set_property description $name $core
+set_property version $version $core
+set_property core_revision 1 $core
+
+ipx::create_xgui_files $core
+ipx::update_checksums $core
+ipx::save_core $core
+ipx::check_integrity $core
+
+ipx::archive_core openhwgroup_risc-v_$name.zip $core
+ipx::unload_core component_1
+
